apps/lasp/src/lasp_core.erl

```diff
@@ -339,15 +346,27 @@ bind(Id, Value, Store) ->
 -spec bind(id(), value(), function(), store()) -> {ok, var()}.
 bind(Id, {delta, Value}, MetadataFun, Store) ->
     Mutator = fun(#dv{type=Type, metadata=Metadata0, value=Value0,
-                      waiting_threads=WT}=Object) ->
+                      waiting_threads=WT, delta_counter=Counter0,
+                      delta_map=DeltaMap0, delta_ack_map=AckMap}=Object) ->
             Metadata = MetadataFun(Metadata0),
             %% Merge may throw for invalid types.
             try
                 Merged = lasp_type:merge(Type, Value0, Value),
                 {ok, SW} = reply_to_all(WT, [], {ok, {Id, Type, Metadata, Merged}}),
-                NewObject = #dv{type=Type, metadata=Metadata, value=Merged, waiting_threads=SW},
-                %% Return value is a delta state
-                {NewObject, {ok, {Id, Type, Metadata, {delta, Value, Merged}}}}
+                case lasp_lattice:is_strict_inflation(Type, Value0, Merged) of
+                    true ->
+                        DeltaMap = orddict:store(Counter0, Value, DeltaMap0),
+                        NewObject = #dv{type=Type, metadata=Metadata, value=Merged,
+                                        waiting_threads=SW,
+                                        delta_counter=increment_counter(Counter0),
+                                        delta_map=DeltaMap, delta_ack_map=AckMap},
+                        %% Return value is a delta state.
+                        {NewObject, {ok, {Id, Type, Metadata, {delta, Value, Merged}}}};
+                    false ->
+                        %% Given delta state is already merged, no delta info update.
+                        {Object#dv{metadata=Metadata, value=Merged, waiting_threads=SW},
```

Comment:
I agree with your comment. I will remove the changes for the value and the metadata.